compute/resource_pipeline.go

```diff
@@ -0,0 +1,231 @@
+}
+
+func (a pipelinesAPI) create(s pipelineSpec, allowDuplicateNames bool) (string, error) {
+	var id createPipelineResponse
+	err := a.client.Post(a.ctx, "/pipelines", s, &id)
+	return id.PipelineID, err
+}
+
+func (a pipelinesAPI) read(id string) (p pipelineInfo, err error) {
+	err = a.client.Get(a.ctx, "/pipelines/"+id, nil, &p)
+	return
+}
+
+func (a pipelinesAPI) update(id string, s pipelineSpec, allowDuplicateNames bool) error {
+	return a.client.Put(a.ctx, "/pipelines/"+id, s)
+}
+
+func (a pipelinesAPI) delete(id string) error {
+	return a.client.Delete(a.ctx, "/pipelines/"+id, map[string]string{})
+}
+
+func (a pipelinesAPI) waitForState(id string, timeout time.Duration, desiredState PipelineState) error {
+	return resource.RetryContext(a.ctx, timeout,
+		func() *resource.RetryError {
+			i, err := a.read(id)
+			if err != nil {
+				return resource.NonRetryableError(err)
+			}
+			state := *i.State
+			if state == desiredState {
+				return nil
+			}
+			return resource.RetryableError(fmt.Errorf("Pipeline %s is in state %s, not yet in state %s", id, state, desiredState))
+		})
+}
+
+func (a pipelinesAPI) waitForDeleted(id string, timeout time.Duration) error {
+	return resource.RetryContext(a.ctx, timeout,
+		func() *resource.RetryError {
+			i, err := a.read(id)
+			if err != nil {
+				if e, ok := err.(common.APIError); ok && e.IsMissing() {
+					return nil
+				}
+				return resource.NonRetryableError(err)
+			}
+			return resource.RetryableError(fmt.Errorf("Pipeline %s is in state %s, not yet deleted", id, *i.State))
+		})
+}
+
+func adjustPipelineResourceSchema(m map[string]*schema.Schema) map[string]*schema.Schema {
+	clusters, _ := m["clusters"].Elem.(*schema.Resource)
+	clustersSchema := clusters.Schema
+	clustersSchema["spark_conf"].DiffSuppressFunc = sparkConfDiffSuppressFunc
+
+	awsAttributes, _ := clustersSchema["aws_attributes"].Elem.(*schema.Resource)
+	awsAttributesSchema := awsAttributes.Schema
+	delete(awsAttributesSchema, "first_on_demand")
+	delete(awsAttributesSchema, "availability")
+	delete(awsAttributesSchema, "spot_bid_price_percent")
+	delete(awsAttributesSchema, "ebs_volume_type")
+	delete(awsAttributesSchema, "ebs_volume_count")
+	delete(awsAttributesSchema, "ebs_volume_size")
+
+	m["libraries"].MinItems = 1
+
+	return m
+}
+
+// ResourcePipeline ...
+func ResourcePipeline() *schema.Resource {
+	var pipelineSchema = common.StructToSchema(pipelineSpec{}, adjustPipelineResourceSchema)
+	return common.Resource{
+		Schema: pipelineSchema,
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			var s pipelineSpec
+			err := common.DataToStructPointer(d, pipelineSchema, &s)
+			if err != nil {
+				return err
+			}
+			api := newPipelinesAPI(ctx, c)
+			id, err := api.create(s, false)
+			if err != nil {
+				return err
+			}
+			if err = api.waitForState(id, d.Timeout(schema.TimeoutCreate), StateRunning); err != nil {
```

Comment:
and can you put `waitForState` into `create` method itself. TF resource would be thinner this way and we can do more integration tests in the future.